Random failures from DeleteAllProfiles unit test

Due to it making direct calls to `docker`, and finding profiles _outside_ the test context.

```
--- FAIL: TestDeleteAllProfiles (0.20s)
    delete_test.go:203: ListProfiles length = 9, expected 8
        valid: []
        invalid: [0xc000d73320 0xc000d73350 0xc000d73380 0xc000d733b0 0xc000d733e0 0xc000d73410 0xc000d73440 0xc000d73470 0xc000d734d0]
```

There should be a mocked docker (`oci.Docker`), similar to the mock `miniHome`.

```go
        pDirs, err := profileDirs(miniHome...)

        cs, err := oci.ListOwnedContainers(oci.Docker)
```

```go
                        err = os.Setenv(localpath.MinikubeHome, td)
```

The invalid profiles are:

[p1 p2_empty_profile_config p3_invalid_profile_config p4_partial_profile_config p5_missing_machine_config p6_empty_machine_config p7_invalid_machine_config p8_partial_machine_config **minikube**]

It also shouldn't just make random calls to Docker like that, but that's another story. (#8577)

--- a/cmd/minikube/cmd/delete_test.go
+++ b/cmd/minikube/cmd/delete_test.go
@@ -194,6 +194,9 @@ func TestDeleteAllProfiles(t *testing.T) {
 		t.Errorf("got %d test machines, expected %d: %s", len(mFiles), numberOfTotalMachineDirs, mFiles)
 	}
 
+	config.DockerContainers = func() ([]string, error) {
+		return []string{}, nil
+	}
 	validProfiles, inValidProfiles, err := config.ListProfiles()
 	if err != nil {
 		t.Error(err)

--- a/pkg/minikube/config/profile.go
+++ b/pkg/minikube/config/profile.go
@@ -192,6 +192,11 @@ func DeleteProfile(profile string, miniHome ...string) error {
 	return os.RemoveAll(ProfileFolderPath(profile, miniPath))
 }
 
+// DockerContainers lists all containers created by docker driver
+var DockerContainers = func() ([]string, error) {
+	return oci.ListOwnedContainers(oci.Docker)
+}
+
 // ListProfiles returns all valid and invalid (if any) minikube profiles
 // invalidPs are the profiles that have a directory or config file but not usable
 // invalidPs would be suggested to be deleted
@@ -203,7 +208,7 @@ func ListProfiles(miniHome ...string) (validPs []*Profile, inValidPs []*Profile,
 		return nil, nil, err
 	}
 	// try to get profiles list based on all containers created by docker driver
-	cs, err := oci.ListOwnedContainers(oci.Docker)
+	cs, err := DockerContainers()
 	if err == nil {
 		pDirs = append(pDirs, cs...)
 	}

--- a/pkg/minikube/config/profile_test.go
+++ b/pkg/minikube/config/profile_test.go
@@ -52,7 +52,15 @@ func TestListProfiles(t *testing.T) {
 		{2, "p5_partial_config", ""},
 	}
 
+	DockerContainers = func() ([]string, error) {
+		return []string{}, nil
+	}
 	val, inv, err := ListProfiles(miniDir)
+
+	num := len(testCasesValidProfs) + len(testCasesInValidProfs)
+	if num != len(val)+len(inv) {
+		t.Errorf("ListProfiles length = %d, expected %d\nvalid: %v\ninvalid: %v\n", len(val)+len(inv), num, val, inv)
+	}
 
 	for _, tt := range testCasesValidProfs {
 		if val[tt.index].Name != tt.expectName {

